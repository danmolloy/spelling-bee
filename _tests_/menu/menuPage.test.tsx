import "@testing-library/jest-dom"
import { screen, render } from "@testing-library/react"
import MenuPage, { MenuPageProps } from "../../components/menu/menuPage"

const mockProps: MenuPageProps = {
  children: <div data-testid="mock-children" />
}

describe("", () => {
  beforeEach(() => {
    render(<MenuPage {...mockProps} />)
  })
  it("menu-page is in the document", () => {
    const menuPage = screen.getByTestId("menu-page")
    expect(menuPage).toBeInTheDocument()
  })
  it("props.children is in the document", () => {
    const children = screen.getByTestId("mock-children")
    expect(children).toBeInTheDocument()
  })
  it("matches snapshot", () => {
    const menuPage = screen.getByTestId("menu-page")
    expect(menuPage).toMatchSnapshot()
  })
  /* it("page closes on blur", () => {})
  it("font matches NYT", () => {})
   */
})

