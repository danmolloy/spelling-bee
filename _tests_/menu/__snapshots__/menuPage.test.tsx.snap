// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` matches snapshot 1`] = `
<div
  class="blur-none flex flex-col z-40 shadow p-6 rounded-2xl w-full md:w-4/5 lg:w-2/3 mt-8 absolute bg-white text-zinc-700 dark:bg-zinc-900 dark:text-zinc-300 dark:border dark:border-zinc-800"
  data-testid="menu-page"
>
  <div
    data-testid="mock-children"
  />
</div>
`;
